pages/visit/events-exhibitions/index.vue

```diff
@@ -278,130 +183,174 @@ const parsedSeriesAndExhibitions = computed(() => {
+    const { past, ...filters } = routeFilters.value
+    const extrafilters = (past !== 'yes')
+      ? [
+          {
+            range: {
+              endDateWithTime: {
+                gte: 'now',
+              },
+            },
+          },
+        ]
+      : []
+    const results = await $dataApi.keywordSearchWithFilters(
+      queryText,
+      config.eventsExhibitionsList.searchFields,
+      'sectionHandle:event OR sectionHandle:exhibition OR sectionHandle:eventSeries',
+      (route.query.filters && JSON.parse(route.query.filters)) ||
+      {},
+      config.eventsExhibitionsList.sortField,
+      config.eventsExhibitionsList.orderBy,
+      config.eventsExhibitionsList.resultFields,
+      config.eventsExhibitionsList.filters,
+      extrafilters,
+    )
+    if (results && results.hits && results.hits.total.value > 0) {
+      console.log('Search ES HITS,', results.hits.hits)
+      hits.value = results.hits.hits
+      noResultsFound.value = false
+    } else {
+      noResultsFound.value = true
+      hits.value = []
+    }
+  } else {
+    hits.value = []
+    noResultsFound.value = false
+  }
+}
+
+function parseHits(hits = []) {
+  return hits.map((obj) => {
+    if (obj._source.sectionHandle === 'event') {
+      return {
+        ...obj._source,
+        to: `/${obj._source.to}`,
+        image: _get(obj._source, 'image[0].image[0]', null),
+        startDate: _get(
+          obj._source,
+          'startDateWithTime',
+          null
+        ),
+        endDate: _get(obj._source, 'endDateWithTime', null),
+        category: _get(
+          obj._source,
+          'eventType[0].title',
+          null
+        ),
+      }
+    } else if (obj._source.sectionHandle === 'exhibition') {
+      return {
+        ...obj._source,
+        to: `/${obj._source.uri}`,
+        image: _get(obj._source, 'image[0].image[0]', null),
+        startDate: _get(obj._source, 'startDate', null),
+        endDate: _get(obj._source, 'endDate', null),
+        category: _get(obj._source, 'sectionHandle', null),
+      }
+    } else if (obj._source.sectionHandle === 'eventSeries') {
+      return {
+        ...obj._source,
+        to: `/${obj._source.uri}`,
+        image: _get(obj._source, 'image[0].image[0]', null),
+        startDate: _get(obj._source, 'startDate', null),
+        endDate: _get(obj._source, 'endDate', null),
+        category: 'Event Series',
+      }
+    } else {
+      return {
+        ...obj._source,
+        to: `/${obj._source.uri}`,
+      }
+    }
+  })
+}
+
+//  This is event handler which is invoked by search-generic component selections
+function getSearchData(data) {
+  console.log('On the page getsearchdata called')
+
+  const filterData =
+    (data.filters && JSON.stringify(data.filters)) || {}
+
+  useRouter().push({
+    path: '/visit/events-exhibitions',
+    query: {
+      q: data.text,
+      filters: filterData,
+    },
+  })
+}
+
+async function setFilters() {
```

Comment:
replace this with
searchFilters.value = [
...getListingFilters(
searchAggsResponse,
config.eventsExhibitionsList.filters
),
{
esFieldName: 'past',
inputType: 'single-checkbox',
label: 'Include Past Events',
},
]

Reply:
Since searchFilters is a ref, hence using searchFilters.value

Reply:
Past events single-checkbox filter needs to be added here, sorry for the confusion in the ticket, I thought we had set up past events in Craft. But it is not a filter that comes from ES.